tardis/visualization/tools/tests/test_sdec_plot.py

```diff
@@ -0,0 +1,294 @@
+            group.create_dataset("_species_list", data=plotter._species_list)
+            group.create_dataset("_keep_colour", data=plotter._keep_colour)
+            # pytest skip this test?
+        else:
+            group = self.hdf_file[subgroup_name]
+            assert (
+                plotter._full_species_list
+                == group.get("_full_species_list").asstr()[()].tolist()
+            )
+            np.testing.assert_allclose(
+                np.asarray(plotter._species_list),
+                group.get("_species_list")[()],
+            )
+            np.testing.assert_allclose(
+                np.asarray(plotter._keep_colour), group.get("_keep_colour")[()]
+            )
+
+    @pytest.mark.parametrize("packets_mode", ["virtual", "real"])
+    @pytest.mark.parametrize("packet_wvl_range", [[500, 9000] * u.AA])
+    @pytest.mark.parametrize("distance", [10 * u.Mpc, 50 * u.Mpc])
+    @pytest.mark.parametrize("nelements", [1, 3])
+    def test_calculate_plotting_data(
+        self,
+        request,
+        plotter,
+        packets_mode,
+        packet_wvl_range,
+        distance,
+        nelements,
+    ):
+        plotter._calculate_plotting_data(
+            packets_mode, packet_wvl_range, distance, nelements
+        )
+
+        # each group is a different combination of arguments
+        subgroup_name = request.node.callspec.id
+        ref_data_path = ""
+        # TODO: get refdata path
+
+        if self.save:
+            # TODO: delete object at ref_data_path
+
+            with h5py.File("sdec_ref.h5", "a") as file:
+                group = file.create_group(subgroup_name)
+                group.attrs["packet_wvl_range"] = packet_wvl_range.cgs.value
+                group.attrs["packets_mode"] = packets_mode
+                group.attrs["distance"] = distance.cgs.value
+                group.attrs["nelements"] = nelements
+
+                group.create_dataset(
+                    "plot_frequency_bins",
+                    data=plotter.plot_frequency_bins.cgs.value,
+                )
+                group.create_dataset(
+                    "plot_wavelength", data=plotter.plot_wavelength.cgs.value
+                )
+                group.create_dataset(
+                    "plot_frequency", data=plotter.plot_frequency.cgs.value
+                )
+                group.create_dataset(
+                    "packet_wvl_range_mask", data=plotter.packet_wvl_range_mask
+                )
+                group.create_dataset(
+                    "emission_species", data=plotter.emission_species
+                )
+                group.create_dataset(
+                    "absorption_species", data=plotter.absorption_species
+                )
+                group.create_dataset(
+                    "modeled_spectrum_luminosity",
+                    data=plotter.modeled_spectrum_luminosity.cgs.value,
+                )
+                group.create_dataset("lum_to_flux", data=plotter.lum_to_flux)
```

Comment:
Since this is scalar, what does this yield in HDF?

Reply:
Thanks for pointing this out, `lum_to_flux` is a `quantity` if `distance` is provided as an argument in the `generate_plot` function. Although `h5py` seems to convert all quantities to arrays on its own, I thought changing quantities to cgs values before would be a safer option. I just forgot to change this.

Reply:
I see